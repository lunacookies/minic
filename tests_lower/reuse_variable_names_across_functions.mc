func a {
	x := 5
}

func b {
	x0 := 5
	x := &x0
}

func c {
	x1 := 5
	x0 := &x1
	x := &x0
}