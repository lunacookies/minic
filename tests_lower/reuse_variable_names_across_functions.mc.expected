func a
	var x i64
	{
		set x = 5
	}

func b
	var x0 i64
	var x *i64
	{
		set x0 = 5
		set x = &(x0)
	}

func c
	var x1 i64
	var x0 *i64
	var x **i64
	{
		set x1 = 5
		set x0 = &(x1)
		set x = &(x0)
	}
tests_lower:10..16: warning: unused variable
tests_lower:39..47: warning: unused variable
tests_lower:81..89: warning: unused variable
